neqo-transport/src/connection/mod.rs

```diff
@@ -2309,7 +2310,13 @@ impl Connection {
                 self.write_appdata_frames(builder, &mut tokens);
             } else {
                 let stats = &mut self.stats.borrow_mut().frame_tx;
-                self.crypto.write_frame(space, builder, &mut tokens, stats);
+                self.crypto.write_frame(
+                    space,
+                    builder,
+                    space == PacketNumberSpace::Initial && self.role == Role::Client,
```

Comment:
I wonder whether we shouldn't just shuffle all the CRYPTO frames.

OR..  the thought just occurred  ... we now know if we have a ClientHello, so you could drop this entirely.  Have the split function return None if it isn't ClientHello and then not shuffle unless it is a ClientHello.  Then you don't need to pass down the parameter at all.